Envfile

```diff
@@ -15,6 +15,9 @@ variable :APP_PROTOCOL, :String, default: "http://"
 variable :COMMUNITY_COPYRIGHT_START_YEAR, :String, default: "2016"
 variable :COMMUNITY_NAME, :String, default: "DEV(local)"
 
+# Email related variables
+variable :DEFAULT_EMAIL, :String, default: "yo@dev.to"
```

Comment:
Hmm It feels super weird that we're setting a default email in the environment and also in the site config. Is there no way we can use it from the Site config, or should we take it out of the site config and use the default email from the ENV (in that case we'll still be keeping alternate emails in the site config)? Just throwing some thoughts going through my head out.

Reply:
You are correct. I confused myself. Let me updates this quickly.

Reply:
You cannot use `SiteConfig` in initializers. If you try to do that you'll get the following error:

```ruby
You cannot use settings before Rails initialize. (RuntimeError)
```

How about updating `SiteConfig` like this:

```ruby
  field :email_addresses, type: :hash, default: {
    default: ENV["DEFAULT_EMAIL"],
    business: "partners@dev.to",
    privacy: "privacy@dev.to",
    members: "members@dev.to"
  }
```

Not sure if that makes more sense though...

Reply:
Yeah SiteConfig can be finicky, only took down the app a couple of times for us to figure that one out 🙃 

@citizen428 I love the idea of making the ENV variable the default!

Reply:
Making the ENV variable the default is a neat idea, however, I can easily see it getting out of sync and confusing the users. 
For example, a user updates their default email in the site config variable and thinks that the default email is changed throughout the site, hence they expect the emails to be sending from the new default email address that they've set in the site config.

I'm leaning towards less ambiguity by in keeping it set in one place with the drawback that its isolated from the other additional emails. i.e. setting the default email in the ENV variable and changing the Site config section to read Additional emails with a note that states that the default email is set as an ENV variable.

Reply:
In addition, we can make the default email in the site config a readonly field so that they can see what the value is set to.

Reply:
^^ _updated my comment_

Reply:
@Ridhwana Good call!

I now did the following: 

1. Change the internal mod page:
    <img width="1078" alt="Screen Shot 2020-06-04 at 14 02 07" src="https://user-images.githubusercontent.com/47985/83726542-dbf2f400-a66d-11ea-905c-6d0dcfd38cf6.png">
2. Updated the params for the controller to not accept the default email:
    ```ruby
    SiteConfig.email_addresses.except(:default).keys
    ```
3. Default `SiteConfig.email_addresses[:default]` to the ENV var:
    ```ruby
    field :email_addresses, type: :hash, default: {
      default: ApplicationConfig["DEFAULT_EMAIL"],
      business: "partners@dev.to",
      privacy: "privacy@dev.to",
      members: "members@dev.to"
    }
```

Theoretically, you could still set the value in a Rails console with `SiteConfig.email_addresses[:default] = ...`, but preventing that would be too hacky. 

Better?

Reply:
@citizen428 I like this much better 🎉. I have a question follow up question - why are we keeping the default email in the Site Config Hash, could we not remove it completely and just use the ENV variable wherever we referencing the default email address like we doing now in show.html.erb ?

Reply:
@Ridhwana I could pretend that I really like the consistency of having all the emails in one place, but to be honest I did a quick grep and was too lazy to make all the changes right now.

I think what we currently have is safe enough™ and with the ongoing generalization work, we may completely remove all default values anyway.

Reply:
Lol kudos for your honestly 😂 This sounds fine to me though, we can make it an iterative approach :)